src/concurrency.md

```diff
@@ -33,6 +34,40 @@ fn main() {
 }
 ```
 
+[ex-rayon-any-all]: #ex-rayon-any-all
+<a name="ex-rayon-any-all"></a>
+## Check the elements of an array against a predicate in parallel
+
+[![rayon-badge]][rayon] [![cat-concurrency-badge]][cat-concurrency]
+
+The `rayon` crate provides a [`ParallelIterator`] trait which includes parallelized implementations of many of the methods from the standard library's [`Iterator`] trait.
+
+This example demonstrates using the [`rayon::any`] and [`rayon::all`] methods. [`rayon::any`] checks in parallel whether any element of the iterator matches the predicate, and returns as soon as one is found. [`rayon::all`] checks in parallel whether all elements of the iterator match the predicate, and returns as soon as a non-matching element is found.
+
+```rust
+extern crate rayon;
+
+use rayon::prelude::*;
+
+fn main() {
+    let mut vec = vec![2, 4, 6, 8];
+
+    let any_odd = vec.par_iter().any(|n| (*n % 2) != 0);
+    assert!(!any_odd);
+
+    let all_even = vec.par_iter().all(|n| (*n % 2) == 0);
+    assert!(all_even);
+
```

Comment:
how about we add a little flavor to the second predicate like some inequality?
```rust
 assert!(vec.par_iter().all(|n| *n < 10));
  vec.push(13);
```

Reply:
It seems like this loses some of the power of the example to demonstrate the relationship between the methods (`when not any are odd then all are even, when any are odd then not all are even`).